src/network/requestQueue/index.js

```diff
@@ -67,6 +84,18 @@ module.exports = class RequestQueue {
     }
   }
 
+  maybeThrottle(clientSideThrottleTime) {
+    if (clientSideThrottleTime) {
+      const minimumThrottledUntil = Date.now() + clientSideThrottleTime
+      if (this.throttledUntil < minimumThrottledUntil) {
+        this.throttledUntil = minimumThrottledUntil
+
+        // No need to schedule anything here: We will do that when there's
+        // actual requests that are affected by this throttling.
+      }
```

Comment:
Nitpicky, but I think this would read more clearly as:

```suggestion
      const minimumThrottledUntil = Date.now() + clientSideThrottleTime
      this.throttledUntil = Math.max(minimumThrottledUntil, this.throttledUntil)
```